eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -167,18 +165,17 @@ jobs:
           testResultsFormat: "VSTest"
           mergeTestResults: true
       - task: Palmmedia.reportgenerator.reportgenerator-build-release-task.reportgenerator@4
-        condition: and(succeededOrFailed(), eq(variables['CollectCoverage'], 'true'))
+        condition: and(succeededOrFailed(), eq(variables['CollectCoverage'], 'true'), eq(variables['coverage.collected'], 'true'))
         displayName: Generate Code Coverage Reports
         inputs:
           reports: $(Build.SourcesDirectory)\sdk\${{parameters.ServiceDirectory}}\**\coverage.cobertura.xml
           targetdir: $(Build.ArtifactStagingDirectory)\coverage
-          reporttypes: Cobertura;HtmlInline_AzurePipelines
+          reporttypes: Cobertura
           filefilters: +$(Build.SourcesDirectory)\sdk\${{parameters.ServiceDirectory}}\**
           verbosity: Verbose
       - task: PublishCodeCoverageResults@1
-        condition: and(succeededOrFailed(), eq(variables['CollectCoverage'], 'true'))
+        condition: and(succeededOrFailed(), eq(variables['CollectCoverage'], 'true'), eq(variables['coverage.collected'], 'true'))
         displayName: Publish Code Coverage Reports
         inputs:
           codeCoverageTool: Cobertura
           summaryFileLocation: $(Build.ArtifactStagingDirectory)\coverage\Cobertura.xml
-          reportDirectory: $(Build.ArtifactStagingDirectory)\coverage
```

Comment:
Did you remove this intentionally?

Reply:
Yeah. Just a bit of clean up while I was at it (it didn't have the desired effect).

Reply:
...to note, the idea behind it was to generate a better report than what Azure did because Azure *used* to use an older version of reportgenerator implicitly in the publish step, but turns out it as new now. Somehow, this also caused an "empty" Azure.Core.TestFramework assembly to show up. It didn't impact the overall coverage numbers, but was a waste of space and potentially confusing since I did a bunch of work to exclude it from coverage.